pkg/resource/cluster.go

```diff
@@ -319,6 +320,50 @@ func (cluster Cluster) IsFresh(fetcher Fetcher) (bool, error) {
 	return cluster.cr.ResourceVersion == actual.ResourceVersion, nil
 }
 
+func (cluster Cluster) LoggingConfiguration(fetcher Fetcher) (string, error) {
+	var logYaml string
+	if cluster.Spec().LogConfigMap != "" {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cluster.Spec().LogConfigMap,
+			},
+		}
+		err := fetcher.Fetch(cm)
+		if err != nil {
+			return "", err
+		}
+
+		for _, val := range cm.Data {
```

Comment:
nit: As `cm.Data` is a map type, this will be non-deterministic. If a user misconfigures the configmap, it's possible that the logging configuration will drift across reconciliation loops.

Can we instead require that users use a well known key or set of keys with an order of priority? For example:

```go
if ok, val := cm.Data["logging.yaml"]; ok {
    return `"` + val + `"`
}

if ok, val := cm.Data["logging.json"]; ok {
    return `"` + val + `"`
}

// No well known keys found, default to any empty or default logging configuration.
return "\"{sinks: {stderr: {channels: [OPS, HEALTH], redact: true}}}\""
```

Reply:
Yes we can do that, but then we should add a validation in web hook that if user supplies the logging cm in the crdbcluster spec we check that cm exist that specific keys or not. If not we error out from the webhook itself.  

Also, needs to decide on how many keys we allow, IMO two should be enough one for yaml `logging.yaml` and one for json `logging.json`.

Reply:
After doing some brooding on the matter, here's what I'm thinking: We only support `logging.yaml` as a single key. If that key isn't present, we fallback to the default configuration. There's also an argument for erroring out if `logging.yaml` isn't present, as this would more closely match the behavior of mounting the configmap into a pod and discovering that the `logging.yaml` file isn't present. Either behavior works for me!

This matches up a bit closer to the flags exposed by cockroach, there's only one flag that accepts YAML. JSON, being a subset of YAML, can successfully be provided through that flag.

It also means we don't have to worry about adding in a webhook. In my experience, most Kubernetes misconfigurations are handled by error loops or fallback behavior rather than validation. It would also be very easy for a user to later change or delete the configmap which means that a webhook couldn't provide that much safety for us.

Let me know what you think!